File: elixir/prime_counter.exs
defmodule PrimeCounter do

  def check_prime(test_value) do
      2..ceil(:math.sqrt(test_value))
        |> Enum.all?(fn divider -> rem(test_value, divider) != 0 end)
  end

  def get_prime_count(max_value) do
    number_of_blocks = System.schedulers_online() * 10

    3..(number_of_blocks + 2)
      |> Stream.map(fn start_value -> start_value..max_value//number_of_blocks end)
      |> Task.async_stream(fn block -> Enum.count(block, &check_prime/1) end, ordered: false, timeout: :infinity)
      |> Enum.reduce(1, fn {:ok, value}, aggregater -> aggregater + value end)
  end
end

{max_value, _} = System.argv()
  |> Enum.at(0)
  |> Integer.parse()

prime_count = PrimeCounter.get_prime_count(max_value)
IO.puts("Elixir: #{prime_count} prime numbers existing up to #{max_value}")

File: python/prime_counter.py
import sys
import math
import multiprocessing

class PrimeCounter:

    @staticmethod
    def check_prime(test_value):
        return all(test_value % divider != 0 for divider in range(2, math.ceil(math.sqrt(test_value)) + 1))

    @staticmethod
    def get_prime_count_block(block):
        return [PrimeCounter.check_prime(value) for value in block].count(True)
            
    @staticmethod
    def get_prime_count(max_value):
        number_of_blocks = multiprocessing.cpu_count() * 10

        with multiprocessing.Pool() as pool:
            block_results = list(pool.imap_unordered(PrimeCounter.get_prime_count_block,
                                                     ((range(start_value, max_value + 1, number_of_blocks)) for start_value in range(3, (number_of_blocks + 3)))))

        return sum(block_results, start=1)

if __name__ == '__main__':
    max_value = int(sys.argv[1])
    
    prime_count = PrimeCounter.get_prime_count(max_value)
    print(f"Python: {prime_count} prime numbers existing up to {max_value}")